core/src/edu/cornell/gdiac/bubblegumbandit/controllers/CollisionController.java

```diff
@@ -245,35 +256,159 @@ public void resolveGumCollision(Obstacle bodyA, Obstacle bodyB){
 
         GumModel gum = null;
         Obstacle body = null;
+        EnemyModel enemy = null;
+        TileModel tile = null;
         if (isGumObstacle(bodyA)) {
             gum = (GumModel) bodyA;
             body = bodyB;
+            if (bodyB instanceof EnemyModel) {
+                enemy = (EnemyModel) bodyB;
+            }
         };
         if (isGumObstacle(bodyB)) {
             gum = (GumModel) bodyB;
             body = bodyA;
+            if (bodyA instanceof EnemyModel) {
+                enemy = (EnemyModel) bodyA;
+            }
         };
         if (gum != null && gum.getName().equals("gumProjectile")) {
             // Do this once gum is turning from a projectile to sticky
             gum.setVX(0);
             gum.setVY(0);
             gum.setTexture(bubblegumController.getStuckGumTexture());
             gum.setName("stickyGum");
-            gum.setRadius(gum.getRadius() * 1.5f);
+            gum.setRadius(gum.getRadius());
             // Changing radius resets filter for some reason
             gum.getFilterData().maskBits = MASK_GUM;
             gum.getFilterData().categoryBits = CATEGORY_GUM;
         }
-
+        //0 = horizontal, 1 = vertical, 2 = rightCorner, 3 = leftCorner
+        int orientation = 0;
         if (gum != null && gum.canAddObstacle(body)){
-            WeldJointDef weldJointDef = bubblegumController.createGumJoint(gum, body);
+            if (enemy != null) {
+                if (!gum.onTile()) {
+                    gum.markRemoved(true);
+                    enemy.setGummedTexture();
+                    enemy.setGummed(true);
+                    gummedRobots.addLast(enemy);
+                }
+                else {
+                    enemy.setStuck(true);
+                }
+            }
```

Comment:
What's gummed vs stuck?
Are they mutually exclusive?

Reply:
gummed is when the robot is hit w gum and turned pink. Stuck is when the robot just hits gum that's on a tile and gets stuck. I differentiated them bc I don't want the gum disappearing if it's just stuck and not gummed.